Shared/Services/DeviceService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DeviceService.swift
+//  SyncExtension
+//
+//  Created by Richard Ascanio on 1/18/24.
+//
+
+import Foundation
+import InternxtSwiftCore
+
+struct DeviceService {
+    static var shared = DeviceService()
+    private let backupAPI: BackupAPI = APIFactory.Backup
+    private let decrypt = Decrypt()
+    private let config = ConfigLoader().get()
+
+    public func getAllDevices(deviceName: String?) async throws -> Array<Device> {
+        let devices = try await backupAPI.getAllDevices()
+        var filteredDevices: [Device] = []
+        var currentDevice: Device? = nil
+
+        for device in devices {
```

Comment:
A simple map and filter will do the trick here, less verbose, and easier to read